lib/miq_automation_engine/models/miq_ae_git.rb

```diff
@@ -0,0 +1,328 @@
+require 'rugged'
+
+class MiqAeGit
+  attr_accessor :name, :email, :base_name
+  ENTRY_KEYS = [:path, :dev, :ino, :mode, :gid, :uid, :ctime, :mtime]
+  DEFAULT_FILE_MODE = 0100644
+  LOCK_REFERENCE = 'refs/locks'
+
+  def initialize(options = {})
+    raise ArgumentError, "Must specify path" unless options.key?(:path)
+    @path          = options[:path]
+    @email         = options[:email]
+    @name          = options[:name]
+    @bare          = options[:bare]
+    @commit_sha    = options[:commit_sha]
+    @password      = options[:password]
+    @ssl_no_verify = options[:ssl_no_verify] || false
+    @remote_name   = 'origin'
+    @cred          = Rugged::Credentials::UserPassword.new(:username => @name,
+                                                           :password => @password)
+    @base_name     = File.basename(@path)
+    return clone(options[:url]) if options[:url]
+    options[:new] ? create_repo : open_repo
+  end
+
+  def create_repo
+    @repo = @bare ? Rugged::Repository.init_at(@path, :bare) :
+                    Rugged::Repository.init_at(@path)
+    @repo.config['user.name']       = @name  if @name
+    @repo.config['user.email']      = @email if @email
+  end
+
+  def open_repo
+    @repo = Rugged::Repository.new(@path)
+  end
+
+  def delete_repo
+    return false unless @repo
+    @repo.close
+    FileUtils.rm_rf(@path)
+    true
+  end
+
+  def add(hash, commit_sha = nil)
```

Comment:
If path and data are required, how about the following (or does it make the caller unclear).  The param name of hash also doesn't say anything about what it does

```
def add(path, data, commit_sha = nil, default_entry_keys = {})
```

Also, if commit_sha is nil, what happens on line 52?